spec/models/hold_spec.rb

```diff
@@ -39,4 +39,37 @@
       expect(build(:invalid_date_time_hold)).not_to be_valid
     end
   end
+
+  describe 'check_conflicting_rentals' do
+    let(:hold) { create(:hold) }
+    let(:conflicting_rental) { create(:hold_conflicting_rental) }
+    let(:future_rental) { create(:far_future_rental) }
+
+    it 'should cancel a conflicting rental' do
+      conflicting_rental
+      hold.check_conflicting_rentals
+      expect(conflicting_rental.reload).to be_canceled
+    end
+
+    it 'should not affect a future rental' do
+      future_rental
+      expect do
+        hold.check_conflicting_rentals
+      end.not_to change(future_rental, :rental_status)
+    end
+
+    it 'should create a new rental when there is a conflicting rental' do
+      conflicting_rental
+      expect do
+        hold.check_conflicting_rentals
```

Comment:
Might also want to do an assertion for sending an email

     expect do
        some thing that sends an email
     end.to change{ActionMailer::Base.delivery_count}.by(1)